python/doc/examples/data_analysis/distribution_fitting/plot_model_singular_multivariate_distribution.py

```diff
@@ -41,29 +41,76 @@ def draw(dist, Y):
+#
+# .. math::
+#
+#      f(u, v_1, v_2) = (y_1, y_2)
+#
+# where:
+#
+# .. math::
+#
+#      y_1 & = sin(u) / (1 + cos(u)^2) + 0.05 * v_1 \\
+#      y_2 & = sin(u) * cos(u) / (1 + cos(u)^2) + 0.05 * v_2
+#
+# We define the following input random vector:
+#
+# .. math::
+#
+#      U  \sim \cU(-0.85\pi, 0.85\pi) \\
+#      (V_1, V_2)  \sim \cN(\vect{\mu} = \vect{0}, \vect{\sigma} = \vect{1}, Id_2)\\
+#
+# with :math:`U` and :math:`\vect{V})` independent.
+#
+# We define the output random vector :math:`\vect{Y}` as:
+#
+# .. math::
+#
+#      \vect{Y} = f(U, V_1, V_2)
+
 f = ot.SymbolicFunction(
-    ["U", "xi1", "xi2"],
-    ["sin(U)/(1+cos(U)^2)+0.05*xi1", "sin(U)*cos(U)/(1+cos(U)^2)+0.05*xi2"],
+    ["U", "v1", "v2"],
+    ["sin(U)/(1+cos(U)^2)+0.05*v1", "sin(U)*cos(U)/(1+cos(U)^2)+0.05*v2"],
 )
 U = ot.Uniform(-0.85 * m.pi, 0.85 * m.pi)
-xi = ot.Normal(2)
-X = ot.BlockIndependentDistribution([U, xi])
+V = ot.Normal(2)
+X = ot.BlockIndependentDistribution([U, V])
+
+# %%
+# We generate a sample of the output random vector :math:`\vect{Y}` of size :math:`N`.
 N = 200
-Y = f(X.getSample(N))
+sample_Y = f(X.getSample(N))
 
 # %%
-# Estimation by multivariate kernel smoothing.
-multi_ks = ot.KernelSmoothing().build(Y)
-view = viewer.View(draw(multi_ks, Y))
+# We estimate the distribution of the output random vector :math:`\vect{Y}` by multivariate kernel smoothing.
+y_multi_ks = ot.KernelSmoothing().build(sample_Y)
+view = viewer.View(draw(y_multi_ks, sample_Y))
 
 # %%
-# Estimation by empirical beta copula.
-beta_copula = ot.EmpiricalBernsteinCopula(Y, len(Y))
+# Now, we estimate the distribution of :math:`\vect{Y}` splitting the estimation of the marginals
+# from the estimation of the copula:
+#
+# - the marginals are fitted by kernel smoothing,
+# - the copula is fitted using the Bernstein copula factory :class:`~openturns.BernsteinCopulaFactory` that builds
+#   an empirical Bernstein copula.
+#
+# First, we do not specify the bin number :math:`m`. It is equal to the value computed by the default method, which is the
+# LogLikelihood criteria. We get :math:`m=1`, which
+# means that one cell is created: the built copula is diffuse in :math:`[0,1]^2`. The estimated copula is
+# the independent copula.
+empBern_copula = ot.BernsteinCopulaFactory().buildAsEmpiricalBernsteinCopula(sample_Y)
+print('bin number computed m = ', empBern_copula.getBinNumber())
 marginals = [
-    ot.KernelSmoothing().build(Y.getMarginal(j)) for j in range(Y.getDimension())
+    ot.KernelSmoothing().build(sample_Y.getMarginal(j)) for j in range(sample_Y.getDimension())
 ]
-beta_dist = ot.JointDistribution(marginals, beta_copula)
-view = viewer.View(draw(beta_dist, Y))
+y_empBern = ot.JointDistribution(marginals, empBern_copula)
+view = viewer.View(draw(y_empBern, sample_Y))
 
+# %%
+# Now, we specify a bin number equal to the sample size: :math:`m = N` so that the built copula is very close to the sample.
+# In that case, it manages to reproduce its specific feature.
```

Comment:
You may mention that in this case, the EmpiricalBernsteinCopula is a beta copula in the sense of Segers

Reply:
yes